File: Crio/crio_programming_interview_problems-master/ValidPalindrome/ValidPalindrome.cpp
#include <bits/stdc++.h>
using namespace std;

class ValidPalindrome {
    public:
        string isPalindrome(string s) {
            // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
            transform(s.begin(), s.end(), s.begin(), ::tolower);
            string t = "";
            for(auto i:s) {
                if((i>='a' && i<= 'z') || (i>='0' && i<='9'))
                    t+=i;
            }
            int n = t.size();
            for(int i = 0; i < n/2; i++) {
                if(t[i] != t[n-1-i])
                    return "false";
            }
            return "true";
            // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        }
};

int main() {
    string s;
    getline(cin, s);
    string result = ValidPalindrome().isPalindrome(s);
    cout<<result;
    return 0;
}

File: Crio/crio_programming_interview_problems-master/ValidPalindrome/ValidPalindrome.java
import java.io.*;
import java.util.*;

public class ValidPalindrome {
    // Implement your solution by completing the below function
    public String isPalindrome(String s) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        if (s.length() == 0)
            return "true";
        s = s.replaceAll("[^a-zA-Z0-9]", "");
        s = s.toLowerCase();
        for (int i = 0, j = s.length() - 1; i <= j; ++i, j--) {
            if (Character.isAlphabetic(s.charAt(i)) || Character.isDigit(s.charAt(i)))
                if (s.charAt(i) != s.charAt(j))
                    return "false";
        }
        return "true";
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        String line = in.readLine();

        String result = new ValidPalindrome().isPalindrome(line);
        System.out.print(result);
    }
}